scipy/signal/tests/test_signaltools.py

```diff
@@ -3565,6 +3565,13 @@ def test_sosfilt_zi(self, dt):
         _, zf = sosfilt(sos, np.ones(40, dt), zi=zi.tolist())
         assert_allclose_cast(zf, zi, rtol=1e-13)
 
+    def test_dtype_deprecation(self, dt):
```

Comment:
An obvious annoyance is that this test runs 4 times for each `dt` value, but doesn't consume `dt`.

Not sure if that meets the threshold to remove it from the parametrized class abstraction or not.

Reply:
Yeah. Did not pass my threshold, the convenience of having all sosfilt testing in a class won over, fwiw.
Can rework, of course, if you've a strong opinion to the contrary?